File: spring-boot-cdc-event-streaming/spring-boot-cdc-event-streaming-mastodon-consumer/src/main/java/com/onlyteo/sandbox/model/SearchResultDto.java
package com.onlyteo.sandbox.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotNull;

import java.util.List;

public record SearchResultDto(@JsonProperty("accounts") @NotNull List<AccountDto> accounts,
                              @JsonProperty("statuses") @NotNull List<StatusDto> statuses) {
}

File: spring-boot-cdc-event-streaming/spring-boot-cdc-event-streaming-mastodon-consumer/src/main/kotlin/com/onlyteo/sandbox/model/SearchResultDto.kt
package com.onlyteo.sandbox.model

import com.fasterxml.jackson.annotation.JsonProperty
import jakarta.validation.constraints.NotNull

data class SearchResultDto(
    @JsonProperty("accounts") @field:NotNull val accounts: List<AccountDto>,
    @JsonProperty("statuses") @field:NotNull val statuses: List<StatusDto>
)
